L1Trigger/Phase2L1ParticleFlow/interface/taus/L1HPSPFTauEmulator.h

```diff
@@ -0,0 +1,426 @@
+#ifndef L1HPSPFTAUEMULATOR_H
+#define L1HPSPFTAUEMULATOR_H
+
+#include "ap_int.h"
+#include "ap_fixed.h"
+#include <iostream>
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include "DataFormats/L1TParticleFlow/interface/taus.h"
+#include "DataFormats/L1TParticleFlow/interface/puppi.h"
+
+namespace L1HPSPFTauEmu {
+
+  //mapping the eta phi onto bits
+
+  static float etaphi_base = 720. / M_PI;
+  static float dz_base = 0.05;
+  typedef l1ct::pt_t pt_t;
+  typedef l1ct::glbeta_t etaphi_t;
+
+  typedef ap_int<13> detaphi_t;
+
+  typedef ap_uint<20> detaphi2_t;
+  typedef ap_uint<5> count_t;  //type for multiplicity
+  typedef ap_uint<3> type_t;   //type for particle type
+  typedef l1ct::z0_t dz_t;
+
+  class Particle : public l1ct::PuppiObj {
+  public:
+    type_t pID;
+    dz_t tempZ0;
+  };
+
+  class Tau : public l1ct::Tau {
+  public:
+    ap_uint<5> seed_index;
+    pt_t seed_pt;
+    etaphi_t seed_eta;
+    etaphi_t seed_phi;
+    dz_t seed_z0;
+  };
+
+  //constants
+  static const detaphi_t strip_phi = 0.20 * etaphi_base;
+  static const detaphi_t strip_eta = 0.05 * etaphi_base;
+
+  static const pt_t min_leadChargedPfCand_pt = l1ct::Scales::makePtFromFloat(1.);
+  static const detaphi_t isoConeSize = 0.4 * etaphi_base;
+  static const detaphi_t delta_Rclean = 0.4 * etaphi_base;
+  static const dz_t dzCut = 0.4 / dz_base;
+  static const etaphi_t etaCutoff = 2.4 * etaphi_base;
```

Comment:
Use `constexpr` instead of `static const` if possible, where not, remove `static` (since these are in a namespace rather than in a class)